protocol/lib/abci/util_test.go

```diff
@@ -0,0 +1,70 @@
+package abci_test
+
+import (
+	"fmt"
+	"github.com/cometbft/cometbft/libs/log"
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dydxprotocol/v4-chain/protocol/lib/abci"
+	"github.com/dydxprotocol/v4-chain/protocol/mocks"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestRunCached_Mixed(t *testing.T) {
+	testEvent := sdk.NewEvent("test", sdk.NewAttribute("key", "value"))
+	tests := map[string]struct {
+		f             func(ctx sdk.Context) error
+		expectedError error
+	}{
+		"success": {
+			f: func(ctx sdk.Context) error {
+				ctx.EventManager().EmitEvent(testEvent)
+				return nil
+			},
+		},
+		"failure": {
+			f: func(ctx sdk.Context) error {
+				ctx.EventManager().EmitEvent(testEvent)
+				return fmt.Errorf("failure")
+			},
+			expectedError: fmt.Errorf("failure"),
+		},
+		"panic": {
```

Comment:
@teddyding please see this unit test to validate that panics do not propagate upwards.